src/Illuminate/Support/Collection.php

```diff
@@ -296,6 +296,34 @@ public function containsStrict($key, $value = null)
         return in_array($key, $this->items, true);
     }
 
+    /**
+     * Join all items from the collection using a string. The final items can use a separate glue string.
+     *
+     * @param  string  $glue
+     * @param  string  $finalGlue
+     * @return string
+     */
+    public function join($glue, $finalGlue = '')
+    {
+        if ($finalGlue === '') {
+            return $this->implode($glue);
+        }
+
+        if ($this->count() === 0) {
+            return '';
+        }
+
+        if ($this->count() === 1) {
+            return $this->last();
```

Comment:
Can this be cast to string explicitly?

Reply:
As my understanding, this one can not type cast to string. `return` type should be mixed.